// @pdfme/manipulator/dist/esm/__tests__/test-helpers.js
import fs from 'fs';
import path from 'path';
import { PDFDocument } from '@pdfme/pdf-lib';
import { pdf2img } from '@pdfme/converter';
export const createTestPDF = async (pageCount) => {
    const pdfDoc = await PDFDocument.create();
    for (let i = 0; i < pageCount; i++) {
        const page = pdfDoc.addPage([500, 500]);
        page.drawText(`Page ${i + 1}`, {
            x: 50,
            y: 450,
            size: 20,
        });
    }
    return pdfDoc.save();
};
export const pdfToImages = async (pdf) => {
    const arrayBuffers = await pdf2img(pdf, { imageType: 'png' });
    return arrayBuffers.map((buf) => Buffer.from(new Uint8Array(buf)));
};
export const getPDFPageCount = async (pdf) => {
    const pdfDoc = await PDFDocument.load(pdf);
    return pdfDoc.getPageCount();
};
export const assetPath = (fileName) => path.join(__dirname, 'assets/pdfs', fileName);
export function toArrayBuffer(buf) {
    return new Uint8Array(buf);
}
export const loadTestPDF = (fileName) => {
    return toArrayBuffer(fs.readFileSync(assetPath(fileName)));
};
