// @pdfme/manipulator/dist/cjs/__tests__/test-helpers.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.loadTestPDF = exports.assetPath = exports.getPDFPageCount = exports.pdfToImages = exports.createTestPDF = void 0;
exports.toArrayBuffer = toArrayBuffer;
const fs_1 = __importDefault(require("fs"));
const path_1 = __importDefault(require("path"));
const pdf_lib_1 = require("@pdfme/pdf-lib");
const converter_1 = require("@pdfme/converter");
const createTestPDF = async (pageCount) => {
    const pdfDoc = await pdf_lib_1.PDFDocument.create();
    for (let i = 0; i < pageCount; i++) {
        const page = pdfDoc.addPage([500, 500]);
        page.drawText(`Page ${i + 1}`, {
            x: 50,
            y: 450,
            size: 20,
        });
    }
    return pdfDoc.save();
};
exports.createTestPDF = createTestPDF;
const pdfToImages = async (pdf) => {
    const arrayBuffers = await (0, converter_1.pdf2img)(pdf, { imageType: 'png' });
    return arrayBuffers.map((buf) => Buffer.from(new Uint8Array(buf)));
};
exports.pdfToImages = pdfToImages;
const getPDFPageCount = async (pdf) => {
    const pdfDoc = await pdf_lib_1.PDFDocument.load(pdf);
    return pdfDoc.getPageCount();
};
exports.getPDFPageCount = getPDFPageCount;
const assetPath = (fileName) => path_1.default.join(__dirname, 'assets/pdfs', fileName);
exports.assetPath = assetPath;
function toArrayBuffer(buf) {
    return new Uint8Array(buf);
}
const loadTestPDF = (fileName) => {
    return toArrayBuffer(fs_1.default.readFileSync((0, exports.assetPath)(fileName)));
};
exports.loadTestPDF = loadTestPDF;
